pip/index.py

```diff
@@ -500,7 +497,7 @@ def mkurl_pypi_url(url):
                 )
 
             raise DistributionNotFound(
-                'No distributions matching the version for %s' % req
+                'No matching distribution found for %s' % req
```

Comment:
Well this is due to the fix of #2502 which merges:

    raise DistributionNotFound('No distributions at all found for %s' % req)

and 

    raise DistributionNotFound('No distributions matching the version for %s' % req)

but I'll reorganize things in 2 differents PRs